src/amqp/publisher.ts

```diff
@@ -13,24 +13,17 @@ export class AMQPPublisher {
   }
 
   public async publish(exchange: string, routingKey: string, data: any): Promise<void> {
-    let promise: PromiseLike<amqp.Channel>;
-    if (this.channel) {
-      promise = Promise.resolve(this.channel);
-    } else {
-      promise = this.connection.createChannel();
+    const channel = await this.getOrCreateChannel();
+    await channel.assertExchange(exchange, 'topic', { durable: false, autoDelete: false });
+    await channel.publish(exchange, routingKey, Buffer.from(JSON.stringify(data)));
+    this.logger('Message sent to Exchange "%s" with Routing Key "%s" (%j)', exchange, routingKey, data);
+  }
+
+  private async getOrCreateChannel(): Promise<amqp.Channel> {
+    if (!this.channel) {
+      this.channel = await this.connection.createChannel();
+      this.channel.on('error', (err) => { this.logger('Publisher channel error: "%j"', err); });
```

Comment:
Generally it would be good to re-establish/retry the channel on error, but felt modifying that functionality was out of scope of this work so I added it to debugging logging to help triage issues. We use another library for other RabbitMQ functionality called [Rascal](https://github.com/guidesmiths/rascal) which does this very well.

Created https://github.com/Surnet/graphql-amqp-subscriptions/issues/16 to address at a later date.